File: 2022/01/11/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumRootToLeaf(root *TreeNode) int {
	return dfs(root, 0)
}

func dfs(node *TreeNode, b int) int {
	if node == nil {
		return 0
	}
	b = b*2 + node.Val
	if node.Left == node.Right {
		return b
	}
	return dfs(node.Left, b) + dfs(node.Right, b)
}

File: 2022/01/11/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        def dfs(node, b):
            if node == None:
                return 0
            b = b * 2 + node.val
            if node.left == node.right:
                return b
            return dfs(node.left, b) + dfs(node.right, b)
        return dfs(root, 0)
